### PR TITLE
docs: Add test specifications for `AccountAllowanceDeleteTransaction`

### LINKED ISSUE #27: Document E2E tests: `AccountAllowanceDeleteTransaction`
Document the E2E `AccountAllowanceDeleteTransaction` tests in `test-specifications/`.

### PR DESCRIPTION
**Pull request by rwalworth**
**Description:**
This PR adds the test specifications for `AccountAllowanceDeleteTransaction` fields.

**Related Issue(s):**

Fixes #27

**Notes for Reviewer(s):**

This PR contains work in #279 so that should be reviewed and merged first.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: test-specifications/crypto-service/accountApproveAllowanceTransaction.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# AccountAllowanceApproveTransaction - Test specification
```
Comment (0xivanov) [inline test-specifications/crypto-service/accountApproveAllowanceTransaction.md:1]: Does this file need to be in this PR? I believe it's already implemented in separate PR.

File: test-specifications/crypto-service/accountAllowanceDeleteTransaction.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+| 21      | Approves an NFT allowance to a spender account from an owner account after already granting an NFT allowance to another account                               | allowances=[{ownerAccountId=<CREATED_OWNER_ID>, spenderAccountId=<VALID_ACCOUNT_ID>, token.tokenId=<FUNGIBLE_TOKEN_ID>, nft.serialNumbers=[<NFT_SERIAL_1>, <NFT_SERIAL_2>, <NFT_SERIAL_3>]}], commonTransactionParams.signers=[<CREATED_OWNER_PRIVATE_KEY>]                                                                            | The allowance approval succeeds and the new spender account has the three NFT allowances and the old spender account has 0. | N                 |
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "method": "approveAllowance",
```
Comment (0xivanov) [inline test-specifications/crypto-service/accountAllowanceDeleteTransaction.md:208]:
```suggestion
  "method": "deleteAllowance",
```

File: test-specifications/crypto-service/accountAllowanceDeleteTransaction.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        "ownerAccountId": "0.0.53232",
+        "spenderAccountId": "0.0.8532",
+        "nft": {
+          "tokenId": "0.0.573298",
+          "serialNumbers": [
+            "123",
+            "456",
+            "789"
+          ],
+          "approvedForAll": true,
```
Comment (0xivanov) [inline test-specifications/crypto-service/accountAllowanceDeleteTransaction.md:221]: I don't think this is needed.

File: test-specifications/crypto-service/accountAllowanceDeleteTransaction.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+  "id": 99232,
+  "result": {
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **ApproveTokenAllowance:**
+
+- Approves an allowance of a token to an account.
```
Comment (0xivanov) [inline test-specifications/crypto-service/accountAllowanceDeleteTransaction.md:111]: Approves?

**Review by 0xivanov — COMMENTED** [review]
`AccountAllowanceDeleteTransaction` has several deprecated methods. Also in a lot of places you are saying `approves` and not `deletes`.
Overall this flow should be revisited.

**Comment by rwalworth** [comment]
> `AccountAllowanceDeleteTransaction` has several deprecated methods. Also in a lot of places you are saying `approves` and not `deletes`. Overall this flow should be revisited.

I guess I got lazy with my copy-pasting! `DeleteAllTokenNftAllowances` should be the only function tested as it is the only function provided by this API. I removed the other functions.

There are deprecated methods but these only make modifications to local class variables that don't ever actually get sent to the network. I didn't think these were necessary to test.

File: test-specifications/crypto-service/accountAllowanceDeleteTransaction.md (hunk 4 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountAllowanceDeleteTransaction.md]: Removed this function

File: test-specifications/crypto-service/accountAllowanceDeleteTransaction.md (hunk 3 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountAllowanceDeleteTransaction.md]: Removed this

File: test-specifications/crypto-service/accountAllowanceDeleteTransaction.md (hunk 2 above)
Comment (rwalworth) [reply test-specifications/crypto-service/accountAllowanceDeleteTransaction.md]: Removed this

**Review by ivaylogarnev-limechain — APPROVED** [review]
LGTM

**Pull request merged**